client/lib/post-normalizer/rule-content-detect-surveys.js

```diff
@@ -0,0 +1,45 @@
+/** @format */
+
+/**
+ * External dependencies
+ */
+import { forEach } from 'lodash';
+import i18n from 'i18n-calypso';
+
+export default function detectSurveys( post, dom ) {
+	if ( ! dom ) {
+		throw new Error( 'this transform must be used as part of withContentDOM' );
+	}
+
+	const surveys = dom.querySelectorAll( '.pd-embed' );
+
+	if ( ! surveys ) {
+		return post;
+	}
+
+	forEach( surveys, survey => {
+		// Get survey details
+		const { domain: surveyDomain, id: surveySlug } = JSON.parse(
```

Comment:
JSON.parse can throw, so we probably want a try / catch around this

Reply:
Excellent point. I'll add a try/catch.